test/e2e/json-rpc/switchEthereumChain.spec.js

```diff
@@ -158,7 +158,52 @@ describe('Switch Ethereum Chain for two dapps', function () {
         });
 
         await driver.switchToWindowWithUrl(DAPP_ONE_URL);
+        // Switch to Dapp One and connect it
+        await driver.switchToWindowWithUrl(DAPP_URL);
+        await driver.clickElement({ text: 'Connect', tag: 'button' });
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        await driver.clickElementAndWaitForWindowToClose({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        // Switch to MM and disconnect localhost 8545 for Dapp One
 
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+        await driver.clickElement(
+          '[data-testid ="account-options-menu-button"]',
+        );
+        await driver.clickElement({
+          text: 'All Permissions',
+          tag: 'div',
+        });
+        await driver.clickElementAndWaitToDisappear({
+          text: 'Got it',
+          tag: 'button',
+        });
+        await driver.clickElement({
+          text: '127.0.0.1:8080',
+          tag: 'p',
+        });
+        const editButtons = await driver.findElements('[data-testid="edit"]');
+
+        assert.ok(editButtons.length > 0, 'Edit buttons are available');
+
+        await editButtons[1].click();
+
+        // Disconnect Mainnet
+        await driver.clickElement({
+          text: 'Localhost 8546',
+          tag: 'p',
+        });
+
+        await driver.clickElement('[data-testid="connect-more-chains-button"]');
+        // Switch to Dapp Two
+        await driver.switchToWindowWithUrl(DAPP_ONE_URL);
```

Comment:
thoughts on the connection happening right after line 147? You also had a nice pattern in the other test of deselecting a network during the connection flow which would be great to use here instead of opening the wallet and disconnecting the chain after connecting it

Reply:
Yes, Updated here https://github.com/MetaMask/metamask-extension/pull/27980/commits/bd26be26bb958d6ffe745d6b76ec79d57885a06f